SearchEngine/lambda_SearchEngine/src/offline/CnDictProducer.cc

```diff
@@ -0,0 +1,143 @@
+#include "../../include/CnDictProducer.h"
+#include "../../include/Configuration.h"
+#include <dirent.h>
+#include <iostream>
+#include <fstream>
+
+
+using std::ifstream;
+
+CnDictProducer::CnDictProducer(SplitToolCppJieba * cuttor)
+:_cuttor(cuttor)
+{
+    //初始化文件绝对路径
+    auto _conf = Configuration::GetInstance("");
+    _dictfilepath = _conf->getConfigMap("data") + "/cn_dictionary.dat";
+    _indexfilepath = _conf->getConfigMap("data") + "/cn_index_table.dat";
+    string cnFilePath = _conf->getConfigMap("chinese");
+    DIR *dir;
+    struct dirent *ent;
+    dir = opendir(cnFilePath.c_str());
+    if (dir != NULL) {
+        while ((ent = readdir(dir)) != NULL) {
+            string filename = ent->d_name;
+            if (filename != "." && filename != "..") {
```

Comment:
we should try our best to avoid making our code too deep.
here we could use the below logic to make the code less deep.
```
if (!dir) {
return;
}
while ((ent = readdir(dir)) != NULL) {
  if (filename == "." || filename == "..") {
    return;
  }
  // other logics
}
```